core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -834,6 +837,8 @@ object Parser extends ParserInstances {
     p match {
       case Impl.Map1(p0, f0) =>
         Impl.Map1(p0, AndThen(f0).andThen(fn))
+      case Impl.Fail() => Impl.Fail()
+      case Impl.FailWith(str) => Impl.FailWith(str)
```

Comment:
Can you reuse the same instances with a cast here?

Reply:
I could, but my heuristic is that paser build time is cheap (done once), but parser run time is where we want to optimize, so I didn't bother... but why not...

Reply:
:laughing:, I guess we each persuaded the other. I'm happy either way.